shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -219,27 +221,32 @@ - (instancetype)init {
   self = [super init];
   NSAssert(self, @"Super must not return null on init.");
 
-  if (@available(iOS 9.3, *)) {
-    _delegate.reset([[ObservatoryDNSServiceDelegate alloc] initWithOwner:self]);
+  auto command_line = flutter::CommandLineFromNSProcessInfo();
+  auto settings = flutter::SettingsFromCommandLine(command_line);
```

Comment:
I suspect rather than self-injecting here, we should keep all the injections in FlutterDartProject

Reply:
Sorry, that was vague, I meant I think FlutterDartProject should already be populating these when constructing from the DefaultSettingsForProcess factory etc. See where it's being read in FlutterEngine.mm. The FlutterEngine makes the FlutterObservatoryPublisher. We should let the engine give the publisher an instance of the settings or give it a bool value when init'ing.